apps/meteor/app/lib/server/functions/processWebhookMessage.ts

```diff
@@ -1,10 +1,10 @@
-import type { IMessage, IUser, RequiredField, MessageAttachment } from '@rocket.chat/core-typings';
+import type { IMessage, IUser, RequiredField, MessageAttachment, IRoom } from '@rocket.chat/core-typings';
```

Comment:
Please, do not refactor the function while we are creating a feature. If the refactor is needed, we need to do it in a separate PR, it's really hard to understand what was indeed changed.

Reply:
I had to refactor due to couple of reasons
1. I ran into function size limit eslint error. So I separated the logic to build message and getRoomWithOptionToJoin into separate functions. 
2. I had to loop channels twice -  1. checking whether a message could be sent to the channels or not. 2. loop again to actually send messages.